### PR TITLE
Paintroid-457 : Flutter: Add .ora support while Save/Load

### PR DESCRIPTION
**Pull request by vishad-tyagi**
Ticket 
Paintroid-457 : Flutter: Add .ora support while Save/Load
https://catrobat.atlassian.net/browse/PAINTROID-457

## New Features and Enhancements
- Support for ora format image while saving/loading.(Emulator recording attached)

https://github.com/Catrobat/Paintroid-Flutter/assets/78733360/e0c2c9e2-061b-4776-8d17-890a297a0e15

## Refactorings and Bug Fixes
- Refactorings and Bug Fixes

#### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Paintroid/blob/develop/README.md) and [wiki pages](https://github.com/Catrobat/Catroid/wiki/) of this repository.

- [x] Include the name of the Jira ticket in the PR’s title
- [x] Include a summary of the changes plus the relevant context
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no compiler or linter warnings
- [x] Perform a self-review of the changes
- [x] Verify to commit no other files than the intentionally changed ones
- [x] Include reasonable and readable tests verifying the added or changed behavior
- [x] Confirm that new and existing unit tests pass locally
- [x] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [x] Stick to the project’s gitflow workflow
- [x] Verify that your changes do not have any conflicts with the base branch
- [ ] After the PR, verify that all CI checks have passed
- [ ] Post a message in the *#paintroid* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

### CODE REVIEW

File: packages/io_library/lib/src/io_handler.dart [hunk 1]
```diff
@@ -7,9 +7,13 @@ import 'package:component_library/component_library.dart';
 import 'package:flutter/material.dart';
 import 'package:flutter_riverpod/flutter_riverpod.dart';
 import 'package:io_library/io_library.dart';
+import 'package:io_library/src/usecase/save_as_ora_image.dart';
```
Comment (msesko) [inline packages/io_library/lib/src/io_handler.dart]: should be exported in io_library so no extra import needed

File: packages/io_library/lib/src/io_handler.dart [hunk 2]
```diff
@@ -152,10 +156,67 @@ class IOHandler {
...
+
+    final imgWidth = canvasState.size.width.toInt();
+    final imgHeight = canvasState.size.height.toInt();
+
+    img.Image layer = await convertUiImageToImgImage(canvasState.cachedImage!);
+
+    final oraImage = OraImage(
+      width: imgWidth,
+      height: imgHeight,
+      layers: [layer], // Single layer based on cachedImage
```
Comment (msesko) [inline packages/io_library/lib/src/io_handler.dart]: remove comment

**Review by msesko — CHANGES_REQUESTED** [review]
Hi @vishad-tyagi, good work. please see comments. also i am not sure about the dart ui package you are sometimes importing. could you explain why this is needed? I noticed "ui.Image" where there was only "Image" before.

**Comment by vishad-tyagi** [comment]
Hi @msesko, I have imported dart:ui because it provides lower-level access to the engine's graphical capabilities, mainly used for detailed graphics operations like direct image manipulation. The ui.Image from dart:ui is used for handling image data at a pixel level, whereas the Image widget is a higher-level abstraction used for displaying images in the UI. Importing dart:ui and using ui.Image is necessary when you need precise control over image data beyond what typical widgets offer.

File: packages/io_library/lib/src/io_handler.dart (hunk 1 above)
Comment (vishad-tyagi) [reply packages/io_library/lib/src/io_handler.dart]: I have committed the required changes.

**Review by msesko — CHANGES_REQUESTED** [review]
Ok, I tested this on iphone and android. For me it does not work. I am not able to see the .ora file nor load it. For reference jpg works

**Comment by msesko** [comment]
> Hi @msesko, I have imported dart:ui because it provides lower-level access to the engine's graphical capabilities, mainly used for detailed graphics operations like direct image manipulation. The ui.Image from dart:ui is used for handling image data at a pixel level, whereas the Image widget is a higher-level abstraction used for displaying images in the UI. Importing dart:ui and using ui.Image is necessary when you need precise control over image data beyond what typical widgets offer.

Good thanks for explaining.

**Comment by vishad-tyagi** [comment]
@msesko Have you tested it in an emulator or a physical device? Can you share the workflow of the same because it is working on mine. I will look into the issue.

**Comment by msesko** [comment]
> @msesko Have you tested it in an emulator or a physical device? Can you share the workflow of the same because it is working on mine. I will look into the issue.

I use emulators for android and ios. I did the same thing as you in the video and am unable to save the ora image. Hvae tested again today with same result. @bakicelebi can you please check

**Comment by vishad-tyagi** [comment]
@msesko Please specify the directory being used in the Android emulator. Ensure that images in the Catrobat format are being saved correctly in the designated folder before attempting to save in the ORA format. I recommend using the Documents folder for this purpose. If the issue persists, kindly share a screen recording of the emulator with me. I will look into the issue and work on it. Thank you

**Comment by msesko** [comment]
> @msesko Please specify the directory being used in the Android emulator. Ensure that images in the Catrobat format are being saved correctly in the designated folder before attempting to save in the ORA format. I recommend using the Documents folder for this purpose. If the issue persists, kindly share a screen recording of the emulator with me. I will look into the issue and work on it. Thank you

So I am able to save other formats but not ora. Will test tom again

**Comment by juliajulie95** [comment]
@msesko please check this again

**Comment by juliajulie95** [comment]
Please resolve conflicts

**Comment by vishad-tyagi** [comment]
@juliajulie95 @msesko @bakicelebi Can I continue to contribute even when I am not selected for GSOC ?

**Comment by juliajulie95** [comment]
Sure, we just thought you were gone, because you did not respond.
You can continue to work on your tickets :)

**Comment by vishad-tyagi** [comment]
@juliajulie95  I did not respond because I was not selected for GSOC even though I contributed the most in recent months so I thought that my codes were not up to mark for your project coding patterns. That is why I stopped committing but then I saw one of my branches got merged and I really wanted to contribute so I asked. 
Thank you I will continue to work on my tickets :)

**Pull request closed without merging**